.github/releases/releaser.py

```diff
@@ -0,0 +1,189 @@
+            E.g: current version on main branch: 0.5.3-SNAPSHOT
+                released version: 1.0.0
+                main branch after release: 1.0.1-SNAPSHOT (patch bumped from release)
+"""
+def getReleaseVersions(sementicVersion, releaseType: str) -> Union[str, str]:
+    currentReleaseVersion = None
+    currentReleaseVersion = sementicVersion.finalize_version()
+    if releaseType == 'minor':
+        currentReleaseVersion = currentReleaseVersion.bump_minor()
+    elif releaseType == 'major':
+        currentReleaseVersion = currentReleaseVersion.bump_major()
+
+    nextVersion = currentReleaseVersion.bump_patch()
+    nextVersion = semver.VersionInfo(*(nextVersion.major, nextVersion.minor, nextVersion.patch, "SNAPSHOT"))
+    return currentReleaseVersion, nextVersion
+
+def updateVersions(parser, version: str):
+    updatePackageJson(UI_PACKAGE_JSON, version)
+    updatePackageJson(RELEASE_PACKAGE_JSON, version)
+    updatePomWithNewVersion(parser, PARENT_POM, version, False)
+    for pom in POM_SOURCES_PATH:
+        updatePomWithNewVersion(parser, pom, version, True)
+
+def main(releaseType: str, justPrint: bool, setToSnapshot: bool):
+    parser = ET.XMLParser(remove_comments=False)
+    currentVersion = getCurrentVersion(parser, PARENT_POM)
+    sementicVersion = semver.VersionInfo.parse(currentVersion)
+
+    currentReleaseVersion, nextVersion = getReleaseVersions(sementicVersion, releaseType)
+    if justPrint:
+        # we print the current version and exit
+        print(sementicVersion)
+
+    elif setToSnapshot:
+        # this is to set the versions to the next SNAPSHOT version after the
+        # release artifacts are published
+
+        if sementicVersion.prerelease == "SNAPSHOT":
+            print("Root pom version is {}; It is already on SNAPSHOT version".format(sementicVersion))
+            exit(1)
+        nextPatchV = sementicVersion.bump_patch()
+        snapshotVer = semver.VersionInfo(*(nextPatchV.major, nextPatchV.minor, nextPatchV.patch, "SNAPSHOT"))
+        updateVersions(parser, str(snapshotVer))
+
+    else:
+        # set the versions to the next release version and prepare to publish
+        # artifacts
+        print("""
+            Version on main: {}
+            Version released now: {}
+            Updated version on main: {}""".format(sementicVersion, currentReleaseVersion, nextVersion))
+
+        if sementicVersion.prerelease != "SNAPSHOT":
+            print("Root pom version is {}; Can only release from a SNAPSHOT version".format(sementicVersion))
+            exit(1)
+
+        updateVersions(parser, str(currentReleaseVersion))
+        # update release manifests
+        for file in listdir(RELEASE_YAML_DIR):
+            filePath = "{}{}".format(RELEASE_YAML_DIR, file)
+            filaName = file.split(".")[0]
+            updateReleaseYaml(filePath, filaName, str(currentReleaseVersion))
+
+
+if __name__ == "__main__":
```

Comment:
this is great! awesome job with this Shabir!

Reply:
Merci!